### PR TITLE
added pages template

### PR DESCRIPTION
**Pull request by AdjectiveAllison**
# What

Added pages and functions tinygo template for use with wrangler generate.

# Motivation

[Github issue to add template](https://github.com/syumai/workers/issues/51)

I was unable to test yet due to being unsure how to test wrangler generate with an alternative branch or local template. 
I also deleted go.mod and go.sum as go.mod referenced the old example module and the other templates did not have these. I'm unsure if this will affect the template or if it will affect go-chi being automatically installed.

### CODE REVIEW

**Comment by AdjectiveAllison** [comment]
Also I left out main = "./build/worker.mjs" from the wrangler.toml. I assume the [[routes]].mjs file handles this.

File: _templates/cloudflare/pages-tinygo/pages/index.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+  </head>
+  <body>
+    <p>
+      <a href="/api/hello">/api/hello</a>
+    </p>
+    <p>
+      <a href="/api/hello?name=syumai">/api/hello?name=Example</a>
```
Comment (syumai) [inline _templates/cloudflare/pages-tinygo/pages/index.html:11]: I want to remove the value `syumai` from this template
```suggestion
      <a href="/api/hello?name=Example">/api/hello?name=Example</a>
```

Comment (AdjectiveAllison) [reply _templates/cloudflare/pages-tinygo/pages/index.html]: Sorry about this! I thought I had changed it! should be good now.

**Review by syumai — APPROVED** [review]
@AdjectiveAllison Thanks!

**Pull request merged**